tests/js/middleware-caseholder.js

```diff
@@ -0,0 +1,134 @@
+// Test `gpii.schema.middleware` and its subcomponents.
+//
+"use strict";
+var fluid        = fluid || require("infusion");
+var gpii         = fluid.registerNamespace("gpii");
+var jqUnit       = require("jqUnit");
+
+require("./lib/sequencer");
+
+// We use just the request-handling bits of the kettle stack in our tests, but we include the whole thing to pick up the base grades
+require("../../node_modules/kettle");
+require("../../node_modules/kettle/lib/test/KettleTestUtils");
+
+// The server-side libraries we are testing
+require("../../");
+
+fluid.defaults("gpii.schema.tests.middleware.request", {
+    gradeNames: ["kettle.test.request.http"],
+    path:       "/gated",
+    port:       "{testEnvironment}.options.port",
+    method:     "POST"
+});
+
+fluid.registerNamespace("gpii.schema.tests.middleware.caseHolder");
+gpii.schema.tests.middleware.caseHolder.examineResponse = function (response, body, shouldBeValid) {
+    var expectedStatus = shouldBeValid ? 200 : 400;
+    jqUnit.assertDeepEq("The status code should be as expected...", expectedStatus, response.statusCode);
+
+    if (!shouldBeValid) {
+        var jsonData = typeof body === "string" ? JSON.parse(body) : body;
+        jqUnit.assertTrue("There should be either document or field errors...", jsonData.documentErrors.length > 0 || Object.keys(jsonData.fieldErrors).length > 0);
+    }
+};
+
+// Wire in an instance of kettle.requests.request.http for each test and wire the check to its onError or onSuccess event
+fluid.defaults("gpii.schema.tests.middleware.caseHolder", {
+    gradeNames: ["fluid.test.testCaseHolder"],
+    mergePolicy: {
+        rawModules:    "noexpand",
+        sequenceStart: "noexpand"
+    },
+    moduleSource: {
+        funcName: "gpii.schema.tests.addRequiredSequences",
+        args:     ["{that}.options.sequenceStart", "{that}.options.rawModules"]
+    },
+    sequenceStart: [
```

Comment:
Yes, I can make the micromodule part of the express test suite and provide a grade that includes the standard wiring.